### PR TITLE
Enable vault configuration via environment variables

### PR DESCRIPTION
**Pull request by UkuLoskit**
##### SUMMARY
These changes enable specifying all of the `hashi_vault` configuration options via environment variables. This is useful, because having to specify these values on each invocation is quite cumbersome. 

As an added bonus, this also serves as a workaround for https://github.com/ansible/ansible/issues/57414.

##### ISSUE TYPE
- Feature Pull Request

##### COMPONENT NAME
hashi_vault lookup plugin

### CODE REVIEW

File: lib/ansible/plugins/lookup/hashi_vault.py [hunk 1]
```diff
@@ -31,8 +31,12 @@
       default: 'http://127.0.0.1:8200'
     username:
       description: Authentication user name.
+      env:
+        - name: VAULT_USERNAME
     password:
       description: Authentication password.
+      env:
+        - name: VAULT_PASSWORD
```
Comment (bcoca) [inline lib/ansible/plugins/lookup/hashi_vault.py:39]: missing version_added, same for rest

File: lib/ansible/plugins/lookup/hashi_vault.py [hunk 2]
```diff
@@ -130,7 +142,7 @@ class HashiVault:
     def __init__(self, **kwargs):
 
         self.url = kwargs.get('url', ANSIBLE_HASHI_VAULT_ADDR)
-        self.namespace = kwargs.get('namespace', None)
+        self.namespace = kwargs.get('namespace', os.environ.get('VAULT_NAMESPACE'))
```
Comment (bcoca) [inline lib/ansible/plugins/lookup/hashi_vault.py:145]: this is not needed, use `get_option` instead

**Review by bcoca — CHANGES_REQUESTED** [review]

**Comment by darxriggs** [comment]
@UkuLoskit can you resolve the conflicts and incorporate the review comments?

File: lib/ansible/plugins/lookup/hashi_vault.py [hunk 3]
```diff
@@ -31,8 +31,12 @@
       default: 'http://127.0.0.1:8200'
     username:
       description: Authentication user name.
+      env:
+        - name: VAULT_USERNAME
```
Comment (bcoca) [inline lib/ansible/plugins/lookup/hashi_vault.py:35]: i would use HASHIVAULT or some other way to avoid confusing with other 'vaults'

**Pull request closed without merging**